api.go

```diff
@@ -987,3 +988,228 @@ func (app *App) APIGetChallengeSkin() func(c echo.Context) error {
 		})
 	})
 }
+
+type APITokenResponse struct {
+	Token string `json:"token" example:"dfghjdrtjdrfr"`
+}
+
+type APILoginRequest struct {
+	Username string `json:"username" example:"notch"`
+	Password string `json:"password" example:"OIXtjQeASfz"`
+}
+
+// APILogin godoc
+//
+//	@Summary	  	Get a token
+//	@Description	Get a token for login credentials.
+//	@Tags		  		users, auth
+//	@Accept				json
+//	@Produce			json
+//	@Success			200	{object}	APITokenResponse
+//	@Failure			400	{object}	APIError
+//	@Failure			401	{object}	APIError
+//	@Failure			423	{object}	APIError
+//	@Failure			500	{object}	APIError
+//	@Router				/drasl/api/v2/login [post]
+func (app *App) APILogin() func(c echo.Context) error {
+	return func(c echo.Context) error {
+		var req APILoginRequest
+		err := c.Bind(&req)
+		if err != nil {
+			return echo.NewHTTPError(http.StatusBadRequest, "Malformed JSON request")
+		}
+
+		var user User
+		result := app.DB.First(&user, "username = ?", req.Username)
+		if result.Error != nil {
+			if errors.Is(result.Error, gorm.ErrRecordNotFound) {
+				return echo.NewHTTPError(http.StatusUnauthorized, "User not found!")
+			}
+			return result.Error
+		}
+
+		if user.IsLocked {
+			return echo.NewHTTPError(http.StatusLocked, "Account is locked.")
+		}
+
+		passwordHash, err := HashPassword(req.Password, user.PasswordSalt)
+		if err != nil {
+			return err
+		}
+
+		if !bytes.Equal(passwordHash, user.PasswordHash) {
+			return echo.NewHTTPError(http.StatusUnauthorized, "Incorrect password!")
+		}
+
+		browserToken, err := RandomHex(32)
+		if err != nil {
+			return err
+		}
+
+		c.SetCookie(&http.Cookie{
+			Name:     "browserToken",
+			Value:    browserToken,
+			MaxAge:   BROWSER_TOKEN_AGE_SEC,
+			Path:     "/",
+			SameSite: http.SameSiteStrictMode,
+			HttpOnly: true,
+		})
+
+		user.BrowserToken = MakeNullString(&browserToken)
+		app.DB.Save(&user)
+
+		return c.JSON(http.StatusOK, APITokenResponse{Token: user.BrowserToken.String})
```

Comment:
`withAPIToken` needs to be modified to also accept a user's `BrowserToken`.

Reply:
I overlooked this when initially writing the code, I meant to return API token. This is now fixed for both routes.